### DISCUSSION TITLE
A chat bot made with phi 3 - (work in progress)

### DISCUSSION THREAD

**Discussion by elephantpanda** [microsoft/onnxruntime-genai#847] [Show and tell]
Here is a little chatbot I made which I upgraded to use the Phi-3 mini int4 (previously it had used Cerebras). It uses the genai API. 

[![Chat bot](https://img.youtube.com/vi/ikibvTHGSPY/0.jpg)](https://www.youtube.com/watch?v=ikibvTHGSPY)

It can tell stories too, but because it was trained on so much scientific stuff it tends to be a bit serious. But it probably needs a bit more prompt engineering.

Here it is telling a fishy story:

[![Chat bot](https://img.youtube.com/vi/KCqo_XX7T4o/0.jpg)](https://www.youtube.com/watch?v=KCqo_XX7T4o)

**Comment by natke** [comment]
This is so cool @elephantpanda!

**Reply by elephantpanda** [reply]
> This is so cool @elephantpanda!

Thanks! It's made in Unity with C# genai and DirectML. The phi3 int4 model was just the right size and speed to make this viable offline. 
Phi 3 is a little serious - I think it must have not been trained on enough natural conversations like social media etc. But it's OK.
Only things in terms of genai that's holding me back is [this bug](https://github.com/microsoft/onnxruntime-genai/issues/833) which can make it randomly stop working. But hopefully that will be fixed.
And in the meantime I can add new features like having two characters talk to each other.
Also, I got to think of a way to test it on different GPUs.
